src/gt4py/next/program_processors/runners/dace/transformations/map_promoter.py

```diff
@@ -268,46 +265,35 @@ def can_be_applied(
-        # get an empty range from the second map. With an empty range, the temporary
-        # data would never be produced, and any other consumer - either in current
-        # state or in a different state - would get uninitialized data.
-        is_second_map_range_unknown = (second_map_entry.map.range.num_elements() > 0) != True  # noqa: E712 [true-false-comparison]  # SymPy fuzzy bools.
-
-        # Technically the promotion creates an invalid SDFG, going back to the example
-        #  in the doc string, after the promotion, but before the fusion, `a[i]` is
-        #  written `M` times. This is not an issue per see, since each time the same
-        #  value is written and if we fuse the write to `a` disappears, if `a` is
-        #  a single use transient. Thus we have to make sure that we do not write
-        #  into global memory.
-        # We also check if any node other than the second map is consuming the data
-        #  produced by the first map. We consider both the consumers directly connected
-        #  to the temporary node in current state and the nodes consuming this data
-        #  in a different state. If so, and the range of the second map is unknown
-        #  at compile time - because purely symbolic - we cannot promote the first
-        #  map, because it could happen that the range of the map is empty and the
-        #  data - actually needed by other SDFG nodes - is never produced.
-        # NOTE: We could accept the fact that we write into global memory multiple
-        #   times the same value, but we will not do it. Furthermore, we ignore the
-        #   case where `t` can not be removed.
-        for oedge in graph.out_edges(first_map_exit):
-            if not isinstance(oedge.dst, dace_nodes.AccessNode):
+        # It might be that the ranges that we add to the first Map from the second Map
+        #  are empty. Which means that the second Map is never executed and after
+        #  promotion, the first Map that was executed before would no longer be run.
+        #  To prevent that we require that the number of iterations the second Map
+        #  performs, at compile time, is larger than zero.
+        #  It is also important that we have to do this check at compile time.
+        second_map_iterations: Any = second_map_entry.map.range.num_elements()
+        if str(second_map_iterations).isdigit():
+            second_map_iterations = int(str(second_map_iterations))
+            if second_map_iterations <= 0:
                 return False
-            if not oedge.dst.desc(sdfg).transient:
+        elif hasattr(second_map_iterations, "free_symbols"):
+            # According to [issue#2095](https://github.com/spcl/dace/issues/2095) DaCe is quite
+            #  liberal concerning the positivity assumption, but in GT4Py this is not possible.
+            second_map_iterations = second_map_iterations.subs(
+                (
+                    (sym, sympy.Symbol(sym.name, nonnegative=False))
```

Comment:
Can we call `dace.Symbol` to avoid importing sympy? I see that the constructor accepts assumptions as keyword arguments.

```suggestion
                    (sym, dace.Symbol(sym.name, nonnegative=False))
```